### DISCUSSION TITLE
Can I inject a makedown text  from ojs code ?

### DISCUSSION THREAD

**Discussion by kwangkim** [quarto-dev/quarto-cli#1907] [Q&A] [answered]
I understand I can make a dynamic pandoc markdown  using `python` or `r`. 
<https://quarto.org/docs/computations/execution-options.html#inline-code>
But I would like to use `ojs` to make a pandoc markdown cell.

I tried the following code.
````
```{ojs}
//| output: asis
//| echo: false
t=`|1|2|
|-|-|
|3|4|`
mt=md`${t}`
t
```
${t}

${mt}

|1|2|
|-|-|
|3|4|
````
The output is given by

![image](https://user-images.githubusercontent.com/10469752/184756249-30ba6dcf-5726-4261-9a25-7b8b8108cd26.png)

- inline `ojs` code does not inject a **pure text**.  It is contained in <span>.
```html
<span class="ojs-in-a-box-waiting-for-module-import observablehq">
<span>|1|2|
|-|-|
|3|4|</span>
</span>
```
- the output from `md` in `ojs` does not match with a `pandoc` markdown output.

<strong>Is there any way to inject a pure text into markdown cell from `ojs`? </strong>

Or is there any markdown js library which render like a `pandoc`? 
It is ok to make a `pandoc` markdown cell from `ojs`.

**Comment by jjallaire** [comment, marked as answer]
There is no way to render pandoc markdown from within the browser (which is where OJS runs). You can of course use the simple OJS markdown renderer but that might not do everything you want. The closest thing to pandoc is to use [markdown-it](https://github.com/markdown-it/markdown-it) with a bunch of plugins to emulate pandoc features (but you still won't get anything close to real pandoc or obviously the quarto extensions to pandoc.

**Reply by kwangkim** [reply]
Is there any way to use `output:asis` for `ojs`? 

If it is possible, since it just a `pure text`, I think we can use quarto pandoc markdown directly.
Python version works very well.
````
```{python}
#| output: asis
t="""|1|2|
|-|-|
|3|4|"""
print(t)
```
````
![image](https://user-images.githubusercontent.com/10469752/184757970-36077382-e2fd-4d12-8604-754d23bc7aad.png)

I think we just need to make an `ojs` version of `print` function. `console.log` did not work.

````
```{ojs}
//| output: asis
t=`|1|2|
|-|-|
|3|4|`
print(t)
```
````

**Comment by jjallaire** [comment]
OJS is rendered *on the client* (so well after Pandoc has done its thing!) It has a totally different computation model so can't emit markdown that will be seen by Pandoc -- it's rendered in the user's browser after they load the page rather than when you call `quarto render`

**Reply by kwangkim** [reply]
I did not know `ojs` runs on a client side.
I will use `python` for the markdown generation, instead.
Thank you very much for your advice.
